### DISCUSSION TITLE
Apply 16-bit LUT on WritableBitmap returned from RenderImage

### DISCUSSION THREAD

**Discussion by soothingmusic** [fo-dicom/fo-dicom#1490] [Q&A]
I'm struggling to apply 16 bit LUT on a grayscale image rendered as WritableBitmap.

```
private int[] lookup = new int[65535+1];
//....fill the lookup table

var bitmap = _dicomImage.RenderImage().As<WriteableBitmap>();
int width = bitmap.PixelWidth;
int height = bitmap.PixelHeight;
int stride = bitmap.BackBufferStride;
int bytesPerPixel = (bitmap.Format.BitsPerPixel + 7) / 8;

byte* pImgData = (byte*)bitmap.BackBuffer;

for (int row = 0; row < height; row++)
{
    for (int col = 0; col < width; col++)
    {
        //... pixelNewValue = lookup[pixelOldValue]
    }
}
```

How do i apply this LUT correctly to a grayscale image?

**Comment by gofal** [comment]
Since you want to apply the LUT on the ready rendered WriteableBitmap, this question is out of scope of fo-dicom. Keep in mind, that the WriteableBitmap is NOT a 16 bit image any more. It is a RGB-Color image with 8 bits per color channel. So you do not have to fully compute the 65k values of the lookup table, 256 values would be sufficient.
And then within your loop you have to apply 3 values per pixel: the green, the red and the blue value. But you will find more information about how to manipulate WirteableBitmaps if you are searching on MSDN about Bitmap manipulation. Propably there are already some built in functions for you purposes.
